tests/MongoDB.Driver.Examples/CausalConsistencyExamples.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Examples
+{
+    public class CausalConsistencyExamples
+    {
+        [Fact]
+        public void Causal_Consistency_Example_1()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            var client = GetClient();
+
+            string itemsCollectionName = "items";
+            var testDatabase = client.GetDatabase("test");
+            testDatabase.DropCollection(itemsCollectionName);
+
+            // Start Causal Consistency Example 1
+            using (var session1 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+            {
+                var currenDate = DateTime.UtcNow.Date;
+                var items = client.GetDatabase("test", new MongoDatabaseSettings
+                    {
+                        ReadConcern = ReadConcern.Majority,
+                        WriteConcern = new WriteConcern(
+                            WriteConcern.WMode.Majority,
+                            TimeSpan.FromMilliseconds(1000))
+                    })
+                    .GetCollection<BsonDocument>("items");
+
+                items.UpdateOne(session1,
+                    Builders<BsonDocument>.Filter.And(new BsonDocument("sku", 111), new BsonDocument("end", BsonNull.Value)),
+                    Builders<BsonDocument>.Update.Set("end", currenDate));
+
+                items.InsertOne(session1, new BsonDocument
+                {
+                    {"sku", "nuts-111"},
+                    {"name", "Pecans"},
+                    {"start", currenDate}
+                });
+            }
+            //End Causal Consistency Example 1
+
+            var result = testDatabase.GetCollection<BsonDocument>(itemsCollectionName).Find("{}").FirstOrDefault();
+            RemoveIds(new[] { result });
+            result["start"].Should().NotBeNull();
+            result.Remove("start");
+            result.Should().Be("{ \"sku\" : \"nuts-111\", \"name\" : \"Pecans\" }");           
+        }
+
+        [Fact]
+        public void Causal_Consistency_Example_2()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            var client = GetClient();
+
+            string itemsCollectionName = "items";
+            var testDatabase = client.GetDatabase("test");
+            testDatabase.DropCollection(itemsCollectionName);
+
+            using (var session1 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+            {
+                var currenDate = DateTime.UtcNow.Date;
```

Comment:
I think you can remove all this extra code... all that is needed for Example 2 is that the session1 variable exist.

Reply:
I tested removing this code, and we still need to do something (anything) with session1 so that the ClusterTime and OperationTime are established. I used:

    client.GetDatabase("admin").RunCommand<BsonDocument>(session1, "{ ping : 1 }");